### ISSUE TITLE
EventBus.handleException() fails to log exception thrown

### ISSUE THREAD

**Issue by bhamiltoncx** [google/guava#2093]
In Guava 18.0, `EventBus.handleException()` incorrectly logs `exception.getCause()` instead of the underlying `exception`:

      logger.log(Level.SEVERE, "Could not dispatch event: " 
          + context.getSubscriber() + " to " + context.getSubscriberMethod(),
          exception.getCause());

This means any `@Subscribe` method which throws an exception fails to log the most important information of all: the actual exception thrown. (If the exception happened to be wrapping another exception, Guava will log the wrapped exception, but that's usually not the case.)

I believe this should be:

      logger.log(Level.SEVERE, "Could not dispatch event: " 
          + context.getSubscriber() + " to " + context.getSubscriberMethod(),
          exception);

To reproduce, try the following. I would expect the log message to include `This message should appear in the logs`, but it doesn't.

```
% cat GuavaBug.java 
import com.google.common.eventbus.EventBus;
import com.google.common.eventbus.Subscribe;

public class GuavaBug {
  private static class Event { }

  @Subscribe
  public void handleEvent(Event event) {
    throw new RuntimeException("This message should appear in the logs");
  }

  public static void main(String[] args) {
    EventBus eventBus = new EventBus();
    GuavaBug guavaBug = new GuavaBug();
    eventBus.register(guavaBug);
    eventBus.post(new Event());
  }
}

% javac -cp guava-18.0.jar GuavaBug.java
% java -cp guava-18.0.jar:. GuavaBug
Jun 24, 2015 2:30:42 PM com.google.common.eventbus.EventBus$LoggingSubscriberExceptionHandler handleException
SEVERE: Could not dispatch event: GuavaBug@7de26db8 to public void GuavaBug.handleEvent(GuavaBug$Event)
```

**Comment by sarxos** [comment]
Hi, shouldn't this be:

```java
 logger.log(Level.SEVERE, "Could not dispatch event: " 
      + context.getEvent() + " to " + context.getSubscriberMethod(),
      exception.getCause());
```

Instead of:

```java
 logger.log(Level.SEVERE, "Could not dispatch event: " 
      + context.getSubscriber() + " to " + context.getSubscriberMethod(),
      exception.getCause());
```

???

**Comment by bgertzfield** [comment]
It would be fine to include all three in the log, I'm sure.

**Comment by bwzhang2011** [comment]
any update with such issue, very long time since guava 18.0 released.

**Comment by cgdecker** [comment]
This should be fixed in 19.0.
